client/app/components/sections/ManageAvailability/ManageAvailability.js

```diff
@@ -37,14 +39,23 @@ const shouldLoad = (isOpen, prevIsOpen) => isOpen && !prevIsOpen;
    component `isOpen` becomes `true`
 */
 const loadInitialDataIfNeeded = (props, prevProps = null) => {
-  const isOpen = props.winder.isOpen;
-  const prevIsOpen = prevProps && prevProps.winder.isOpen;
+  const isOpen = props.isOpen;
+  const prevIsOpen = prevProps && prevProps.isOpen;
 
   if (shouldLoad(isOpen, prevIsOpen)) {
     props.calendar.onMonthChanged(props.calendar.initialMonth);
   }
 };
 
+const setPushState = (state, title, path) => {
```

Comment:
As mentioned in the long Flowdock discussion, I think there might be some better places to do `window.location` manipulation, e.g.

1. A routing library
1. A non-React module that subscribes to the Store
1. A separate React component that doesn't render anything but only manages the `window.location`.

The problem with `window.location` manipulation in a component such as this is that the `window.location` is a global thing. And when component is manipulating a global thing, it's messing around with stuff that is not inside the component boundaries. What happens if two (or three) components are messing around with the `window.location`? Clashes are very possible. In addition, what happens when this component in rendered in some other context, e.g. in story book? Will it mess the story book URL?

So I'm a bit concerned about doing the hash manipulation in this component. It might be an okeyish solution as "well this time only and when there will be next time, then well refactor", but the story book part is still concerning.